### DISCUSSION TITLE
Separate water heater sensors

### DISCUSSION THREAD

**Discussion by MKBU2** [veista/nilan#104] [Ideas]
The current Nilan integration seems to couple together the Nilan Bottom Water Heater + target and Nilan Bottom Water Heater + target
I think it would make sense to decouple them. It seems that I cant make logical rules in Home Assistant with the current setup.

Any reason why this wouldn't be possible?

**Comment by veista** [comment]
I don't understand the question.

**Comment by MKBU2** [comment]
I would like that the:
"water_heater.nilan_bottom_water_heater"
"water_heater.nilan_top_water_heater"

would only give the current temperature same style that is for example the "sensor.nilan_controller_board_temperature_t0"

Now if I want to make History graph Card it gives always the limit values that I don't need there. They could be separate values and user could select them if one wants.

**Comment by veista** [comment]
You can use templating to create a sensor {{ state_attr('water_heater.nilan_bottom_water_heater', 'current_temperature') |float}} 
and you can use this also like this in automations.

**Comment by MKBU2** [comment]
Ok good. Maybe it could be like that by default so that the integration part wouldn't make any decisions how end user wants to see different variables.

**Reply by veista** [reply]
I've tried to minimize redundant information. Due to the nature of the polling system each sensor calls the physical hardware. This may overload some networks.

This is only a case of users not understanding how HA works.

**Comment by jiripatka** [comment]
I think there is missing configuration boiler heather temperature in file: sensor.py   I tried add here T11 + T 12, but without success - no sensor   . Any next compilation needs ?               
"get_t11_electric_water_heater_temperature": [
        Map(
            "return_hot_water_top_temperature_t11",
            UnitOfTemperature.CELSIUS,
            SensorDeviceClass.TEMPERATURE,
            SensorStateClass.MEASUREMENT,
            None,
            None,
        )
    ],
 "get_t12_compressor_water_heater_temperature": [
        Map(
            "return_hot_water_bottom_temperature_t12",
            UnitOfTemperature.CELSIUS,
            SensorDeviceClass.TEMPERATURE,
            SensorStateClass.MEASUREMENT,
            None,
            None,
        )
    ],

**Reply by veista** [reply]
I think your question is in the wrong place. If you think there is a missing sensor/s, please follow the instructions in read-me.

**Comment by jiripatka** [comment]
Do you mean https://github.com/veista/nilan/blob/master/README.md ? or ?

**Reply by veista** [reply]
Yes. On other Issues:

Submit the following: Logs, Modbus Version, Device Type - as Shown in the Integration, Device Version - as Shown in the Integration

**Reply by jiripatka** [reply]
Logs veista v. 1.12.16   Home Assistant Core 2024.8.1 Frontend 20240809.0
Modbus version CTS602 via Waveshare RS485 TO ETH (B)
device CompactS od Nilan Firmware: 3.6.57 Hardware: 96
Entites: 78, Sensors: 26, 
Sensors water temperature = T11 + T12 are missing - I can see them only through devices water heather top/bottom, but no as entities.

**Reply by veista** [reply]
Ok. This is already anwsered in this thread. Use templating if you for some reason need a separate sensor for this.

**Comment by jiripatka** [comment]
Sorry ( I've only been on HA for a week :-) and thank you, it's working.

pá 16. 8. 2024 v 18:01 odesílatel Taneli Veistinen ***@***.***>
napsal:
